sc2/bot_ai.py

```diff
@@ -304,16 +304,16 @@ def can_afford(self, item_id: Union[UnitTypeId, UpgradeId, AbilityId], check_sup
         return False
 
     def select_build_worker(self, pos: Union[Unit, Point2, Point3], force: bool=False) -> Optional[Unit]:
-        """Select a worker to build a bulding with."""
+        """Select a worker to build a building with."""
 
-        workers = self.workers.closer_than(20, pos) or self.workers
-        for worker in workers.prefer_close_to(pos).prefer_idle:
-            if not worker.orders or len(worker.orders) == 1 and worker.orders[0].ability.id in {AbilityId.MOVE,
-                                                                                                AbilityId.HARVEST_GATHER,
-                                                                                                AbilityId.HARVEST_RETURN}:
-                return worker
+        workers = self.workers.gathering.closer_than(20, pos) | self.workers.idle or self.workers
```

Comment:
Please make one units call out of this by using `self.works.filter`

Reply:
hopefully fixed